api/services/Ban.js

```diff
@@ -95,6 +95,9 @@ exports.updateBanlist = async function (redToken, username, banlistEntry, friend
         ||_.intersection(lines[i].match(/(\d{4}-){2}\d{4}/g), friend_codes).length) {
       // User was an alt account, modify the existing line instead of creating a new one
       let blocks = lines[i].split(/\s*\|\s*/);
+      if (blocks.length === 6 && !(blocks[5])) { // to handle banlist entries with | at the end
```

Comment:
Run me through how this was causing the issue previously? The final block was null? And this was causing an error?

Reply:
We used to have some old entries with the pipe at the end of the line like this:

> username | 1234-1234-1234 | Reason | IGN | No |

When it was splitted here:

    let blocks = lines[i].split(/\s*\|\s*/);

`blocks.length` was 6  and blocks[5] was empty string `""`. Based on that, the line was skipped and new line was added to the banlist without associating with the already existing entry.

By the way, in parallel we work to remove the trailing pipes from the banlist page so this will be no longer necessary but I thought it's good to add this just in case.